modules/wallet.go

```diff
@@ -3,149 +3,257 @@ package modules
-	// FundSiacoins will add a siacoin input of exaclty 'amount' to the
-	// transaction. A parent transaction may be needed to achieve an input with
-	// the correct value. The siacoin input will not be signed until 'Sign' is
-	// called on the transaction builder.
-	FundSiacoins(amount types.Currency) error
-
-	// TODO: Add a function 'FundSiafunds' - maybe best to wait for a clear use
-	// case, but maybe not - the builder is supposed to be general.
-
-	// AddMinerFee adds a miner fee to the transaction, returning the index of
-	// the miner fee within the transaction.
-	AddMinerFee(fee types.Currency) uint64
-
-	// AddSiacoinInput adds a siacoin input to the transaction, returning the
-	// index of the siacoin input within the transaction. When 'Sign' gets
-	// called, this input will be left unsigned.
-	AddSiacoinInput(types.SiacoinInput) uint64
-
-	// AddSiacoinOutput adds a siacoin output to the transaction, returning the
-	// index of the siacoin output within the transaction.
-	AddSiacoinOutput(types.SiacoinOutput) uint64
-
-	// AddFileContract adds a file contract to the transaction, returning the
-	// index of the file contract within the transaction.
-	AddFileContract(types.FileContract) uint64
-
-	// AddFileContractRevision adds a file contract revision to the
-	// transaction, returning the index of the file contract revision within
-	// the transaction. When 'Sign' gets called, this revision will be left
-	// unsigned.
-	AddFileContractRevision(types.FileContractRevision) uint64
-
-	// AddStorageProof adds a storage proof to the transaction, returning the
-	// index of the storage proof within the transaction.
-	AddStorageProof(types.StorageProof) uint64
-
-	// AddSiafundInput adds a siafund input to the transaction, returning the
-	// index of the siafund input within the transaction. When 'Sign' is
-	// called, this input will be left unsigned.
-	AddSiafundInput(types.SiafundInput) uint64
-
-	// AddSiafundOutput adds a siafund output to the transaction, returning the
-	// index of the siafund output within the transaction.
-	AddSiafundOutput(types.SiafundOutput) uint64
-
-	// AddArbitraryData adds arbitrary data to the transaction, returning the
-	// index of the data within the transaction.
-	AddArbitraryData(arb []byte) uint64
-
-	// AddTransactionSignature adds a transaction signature to the transaction,
-	// returning the index of the signature within the transaction. The
-	// signature should already be valid, and shouldn't sign any of the inputs
-	// that were added by calling 'FundSiacoins' or 'FundSiafunds'.
-	AddTransactionSignature(types.TransactionSignature) uint64
-
-	// Sign will sign any inputs added by 'FundSiacoins' or 'FundSiafunds' and
-	// return a transaction set that contains all parents prepended to the
-	// transaction. If more fields need to be added, a new transaction builder
-	// will need to be created.
+		FundType       types.Specifier
+		OutputID       types.OutputID
+		RelatedAddress types.UnlockHash
```

Comment:
It's the address that's relevant to the motion of the funds.

So if it's a 'SiacoinInput', then it's the address that the original output was sent to (mostly a useless thing to know). If it's a 'SiacoinOutput', then it's the address that the siacoin output was sent to.